b2luigi/__init__.py

```diff
@@ -68,3 +69,85 @@ def requires(_self):
         task_that_requires.requires = requires
 
         return task_that_requires
+
+
+class inherits(object):
+
+    """
+    Similar to `b2luigi.requires`, this copies the luigi.inherits functionality but allows specifying parameters you
+    don't want to inherit.
+
+    It can e.g. be used in tasks that merge the output of the tasks they require. These merger tasks don't need
+    the parameter they resolve anymore but should keep the same order of parameters, therefore simplifying the directory
+     structure created by `b2luigi.Task.add_to_output`.
+
+    Usage can be similar to this:
+
+        class TaskA(b2luigi.Task):
+             some_parameter = b2luigi.IntParameter()
+             some_other_parameter = b2luigi.IntParameter()
+
+             def output(self):
+                 yield self.add_to_output("test.txt")
+
+         @b2luigi.inherits(TaskA, without='some_other_parameter')
+         class TaskB(b2luigi.Task):
+             another_parameter = b2luigi.IntParameter()
+
+             def requires(self):
+                for my_other_parameter in range(10):
+                    yield self.clone(TaskA, some_other_parameter=my_other_parameter)
+
+            def run(self):
+                # somehow merge the output of TaskA to create "out.dat"
+                pass
+
+             def output(self):
+                 yield self.add_to_output("out.dat")
+
+    Parameters:
+        without: Either a string or a collection of strings
+
+    """
+
+    def __init__(self, *tasks_to_inherit, without: Optional[Union[Collection[str], str]] = None):
+        super(inherits, self).__init__()
+        if not tasks_to_inherit:
+            raise TypeError("tasks_to_inherit cannot be empty")
+
+        self.tasks_to_inherit = tasks_to_inherit
+        if isinstance(without, str):
+            self.without = [without]
+        elif without is None:
+            self.without = []
+        else:
+            self.without = list(without)
+
+    def __call__(self, task_that_inherits):
+        # Get all parameter objects from each of the underlying tasks
+
+        for task_to_inherit in self.tasks_to_inherit:
+            for param_name, param_obj in task_to_inherit.get_params():
+                # Check if the parameter exists in the inheriting task and isn't in without
+                if not hasattr(task_that_inherits, param_name) and param_name not in self.without:
+                    # If not, add it to the inheriting task
+                    setattr(task_that_inherits, param_name, param_obj)
+                elif param_name in self.without:
+                    self.without.remove(param_name)
+
+        # Make sure we're only removing parameters that exist
+        assert len(self.without) == 0, f"You're trying to remove parameter(s) {self.without} which do(es) not exist."
```

Comment:
More pythonic would be 

```suggestion
        assert not self.without, f"You're trying to remove parameter(s) {self.without} which do(es) not exist."
```

Reply:
Ok sure, that's definitely more pythonic.